cmd/argocd/commands/app.go

```diff
@@ -1617,3 +1619,70 @@ func NewApplicationPatchCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
 	command.Flags().StringVar(&patch, "patch", "", "Patch")
 	return &command
 }
+
+func NewApplicationPatchResourceCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
+	var patch string
+	var patchType string
+	var resourceName string
+	var namespace string
+	var kind string
+	var group string
+	command := cobra.Command{
+		Use:   "patch-resource APPNAME",
+		Short: "Patch resource in an application",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) != 1 {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+			appName := args[0]
+
+			if resourceName == "" {
+				c.HelpFunc()(c, args)
+				log.Fatal("--resourceName is required")
+			}
+
+			conn, appIf := argocdclient.NewClientOrDie(clientOpts).NewApplicationClientOrDie()
+			defer util.Close(conn)
+			ctx := context.Background()
+			resources, err := appIf.ManagedResources(ctx, &application.ResourcesQuery{ApplicationName: &appName})
+			errors.CheckError(err)
+			liveObjs, err := liveObjects(resources.Items)
+			errors.CheckError(err)
+			for _, obj := range liveObjs {
+				gvk := obj.GroupVersionKind()
+				if kind != "" && kind != gvk.Kind {
+					continue
+				}
+				if group != "" && kind != gvk.Group {
+					continue
+				}
+				if namespace != "" && namespace != obj.GetNamespace() {
+					continue
+				}
+				if resourceName == obj.GetName() {
+					_, err := appIf.PatchResource(ctx, &application.ApplicationResourcePatchRequest{
+						Name:         &appName,
+						Namespace:    obj.GetNamespace(),
+						ResourceName: resourceName,
+						Version:      gvk.Version,
+						Group:        gvk.Group,
+						Kind:         gvk.Kind,
+						Patch:        patch,
+						PatchType:    patchType,
+					})
+					errors.CheckError(err)
+					break
+				}
+			}
+		},
+	}
+
+	command.Flags().StringVar(&patch, "patch", "", "Patch")
+	command.Flags().StringVar(&patchType, "patchType", string(types.MergePatchType), "Which Patching strategy to use: 'application/json-patch+json', 'application/merge-patch+json', or 'application/strategic-merge-patch+json'. Defaults to 'application/merge-patch+json'")
+	command.Flags().StringVar(&resourceName, "resourceName", "", "Name of resource")
+	command.Flags().StringVar(&kind, "kind", "", "Kind")
```

Comment:
With kubectl you can use the following syntax:

```
kubectl patch deployment.v1.apps ks-guestbook-ui
```

Can we do the same instead of using `--kind`, `--group` flags. e.g. ?

Reply:
sure

Reply:
I forgot that command requires appName . Please disregard my command since it is either way not consistent with kubectl